lib/models/navigation.dart

```diff
@@ -32,19 +32,42 @@ final GlobalKey<NavigatorState> navKey = GlobalKey<NavigatorState>();
 /// {@macro models.navigator}
 Nav get navigator => Nav(navKey.currentState!);
 
-/// {@macro models.navigator}
+/// {@template models.navigator_example}
+/// Example:
+///
+/// ```dart
+/// void main() {
+///   runApp(const App()); // "/"
+/// }
+///
+/// navigator.push(const Screen1()); // "/Screen1"
+///
+/// navigator.push(const Screen2()); // "/Screen1/Screen2"
+///
+/// navigator.pushReplacement(const Replacement()); // "/Screen1/Replacement"
+///
+/// navigator.pop(); // "/Screen1"
+///
+/// navigator.pop(); // "/"
+/// ```
+/// {@endtemplate}
 extension type Nav(NavigatorState navigator) {
-  /// if you call [push], you'll navigate to a new widget,
-  /// and calling [pop] will take you back to where you came from.
+  /// Adds a new screen to the route.
+  ///
+  /// {@macro models.navigator_example}
   Future<T?> push<T>(Widget destination) =>
       navigator.push<T>(MaterialPageRoute<T>(builder: (context) => destination));
 
-  /// if you call [pushReplacement], you'll navigate to a new widget,
-  /// and calling [pop] won't do anything.
+  /// Adds a new screen in place of the current screen.
+  ///
+  /// {@macro models.navigator_example}
   Future<void> pushReplacement(Widget destination) => navigator.pushReplacement<void, void>(
         MaterialPageRoute<void>(builder: (context) => destination),
       );
```

Comment:
You can use `///` to add documentation to any class or value (except for local function variables).

This lets you do a lot of cool things:
- link to something else using `[brackets]`
- use `{@template}` and `{@macro}` to reference documentation from somewhere else
- add bullet points, code snippets, and more (using [Markdown syntax](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax))

You can see the documentation any time you hover your mouse on the relevant text:

![Dart doc comments](https://github.com/theheartcenter-one/thc/assets/10457200/d8a7dc64-a4fe-4ba2-801a-fa58323d1a10)